terraform/context.go

```diff
@@ -138,9 +138,16 @@ func NewContext(opts *ContextOpts) (*Context, tfdiags.Diagnostics) {
 	// Determine parallelism, default to 10. We do this both to limit
 	// CPU pressure but also to have an extra guard against rate throttling
 	// from providers.
+	// We throw an error in case of negative parallelism or
+	// parallelism equal to 0.
 	par := opts.Parallelism
-	if par == 0 {
-		par = 10
```

Comment:
There's a log of code that depends on the zero value of `opts.Parallelism` triggering the default. 
Adding an error for < 0 should be OK for now, but I would hold off on trying to refactor everything upstream of the `NewContext` call sites to set the option directly.

Reply:
I agree with you, I reverted the default value logic